abel/tools/circularize.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+from __future__ import division
+import numpy as np
+from scipy import ndimage
+from scipy.interpolate import UnivariateSpline
+from scipy.optimize import leastsq
+
+import abel
+
+# see https://github.com/PyAbel/PyAbel/issues/186 for discussion
+
+
+def circularize_image(IM, method="lsq", center=None, radial_range=None,
+                      zoom=1, smooth=0, nslices=32, ref_angle=None, 
```

Comment:
`smooth` is essential. Noisy data will cause the spline to go haywire. Conversely, too much smoothing will not fully undistort the image. It is essential that the user checks the evaluated correction points and the (smoothed) spline.